nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AbstractCommentManager.java

```diff
@@ -160,4 +162,38 @@ protected void setCommentPermissions(CoreSession session, DocumentModel document
         return ancestorIds;
     }
 
+    /**
+     * @since 11.1
+     */
+    protected DocumentModel getThreadForComment(CoreSession session, DocumentModel comment)
+            throws CommentSecurityException {
+
+        NuxeoPrincipal principal = session.getPrincipal();
+        return CoreInstance.doPrivileged(session, s -> {
```

Comment:
I would switch the names `s` and `session` because we tend to naturally want to use `session` in the lambda.

Same in the next method.